kiali/kiali
# Add external controlplane support

### Describe the change

Adds support for the external controlplane deployment model. Also does some cleanup of the istio config service.

### Steps to test the PR

Manual testing:
1. `hack/run-integration-tests.sh --test-suite backend-external-controlplane --setup-only true`
2. Visit "mesh" page.
3. Verify functionality of overview / list pages.

### Automation testing

Automated testing: `hack/run-integration-tests.sh --test-suite backend-external-controlplane`

Added a hack script to setup an external controlplane environment. Added a github workflow to run e2e tests against the external controlplane environment.

Added backend external controlplane tests to verify that the `/mesh` endpoint returns the proper info for the external-controplane deployment.

### Issue reference

Fixes #4729 #6036
